### ISSUE TITLE
ZapierPromise does not use scoped prototype functionality

### ISSUE THREAD

**Issue by codebycaleb** [zapier/zapier-platform#481]
### Bug Description

[ZapierPromise](https://github.com/zapier/zapier-platform/blob/master/packages/core/src/tools/promise.js) does not adhere to bluebird's [scoped prototypes](http://bluebirdjs.com/docs/features.html#scoped-prototypes) recommendations regarding `require`ing bluebird.

### Reproduction Steps

1. Create an app that uses bluebird for promises
2. Create a Zap using the app and test it via the editor

Errors will be logged that cite bluebird's documentation about [not invoking the Promise constructor directly](http://bluebirdjs.com/docs/error-explanations.html#error-the-promise-constructor-cannot-be-invoked-directly).

The code should work without errors.

### Version Info

- Version info:  core: 11.2.0
- App id:  154066

**Comment by xavdid** [comment]
@codebycaleb do you have a screenshot of how dire the error messages look? Trying to gauge severity here. And does this only affect CLI apps which _separately_ use bluebird promises? If so, seems fairly narrow. We were just talking about pulling the bluebird stuff out, but it's esoteric and working just fine, so we're hesitant to touch it if we can avoid it.

**Comment by codebycaleb** [comment]
Yep! It's not mine though. 😅  Here's an obfuscated version: ![](https://cdn.zappy.app/3f53244909cfadac787b152dd2ca7f2f.png)

> And does this only affect CLI apps which separately use bluebird promises?

Yeah I think so! This app had a dependency for `bluebird@^3.7.2`. Definitely agreed on the narrow point - this is the first report I've ever seen and as far as I know this has essentially always been the case. :)

**Comment by xavdid** [comment]
ok, cool! Good to know.

Yeah, it's the first I've heard of it too. We'll keep this open in case other people have the same issue, but it doesn't strike me as a priority.

**Comment by dpovey** [comment]
Just noting I have very similar symptoms but I am not using bluebird in my app directly, it's just imported as a dependency of zapier-platform-core

```
adpulse.io@1.0.0 /Users/dpovey/src/ppcsamurai/zapier-adpulse.io
└─┬ zapier-platform-core@11.2.0
  └── bluebird@3.7.2
```

When I try to test my Zap I get about 20% of the time the error:

<img width="842" alt="image" src="https://user-images.githubusercontent.com/395066/152489125-a89537d0-6f4a-48ab-83a8-fad4cd3e5e61.png">

And then when activating my Zap about a similar percentage:
<img width="427" alt="image" src="https://user-images.githubusercontent.com/395066/152489336-8ff7f46d-3cb3-4aa2-acd3-009af7a8a55d.png">

**Comment by syedfahad1122** [comment]
test

**Comment by eliangcs** [comment]
We removed bluebird in favor of native Promises in [v17](https://github.com/zapier/zapier-platform/blob/main/CHANGELOG.md#1700), so I assume this is no longer an issue. Closing.
